flink-python/pyflink/table/table_environment.py

```diff
@@ -657,19 +657,55 @@ def drop_temporary_table(self, table_path: str) -> bool:
         """
         return self._j_tenv.dropTemporaryTable(table_path)
 
+    def drop_table(self, table_path: str) -> bool:
+        """
+        Drops a table registered in the given path.
+
+        Temporary objects can shadow permanent ones.
+        If a permanent object in a given path exists,
+        it will be inaccessible in the current session.
+        To make the permanent object available again
+        one can drop the corresponding temporary object.
+
+        :param table_path: The path of the registered table.
+        :return: True if a table existed in the given path and was removed.
+
+        .. versionadded:: 2.0.0
+        """
+        return self._j_tenv.dropTable(table_path)
+
     def drop_temporary_view(self, view_path: str) -> bool:
         """
         Drops a temporary view registered in the given path.
 
         If a permanent table or view with a given path exists, it will be used
         from now on for any queries that reference this path.
 
+        :param view_path: The path of the registered temporary view.
         :return: True if a view existed in the given path and was removed.
 
         .. versionadded:: 1.10.0
         """
         return self._j_tenv.dropTemporaryView(view_path)
 
+    def dropView(self, view_path: str) -> bool:
+        """
+
+        Drops a view registered in the given path.
+
+        Temporary objects can shadow permanent ones.
```

Comment:
I am curious, does this work in all permutations, can we have a temporary table shadowing a permanent view? Have we a test for this? And amend the text if it does not work in all permutations.

Reply:
So for this javadoc for dropView; the view is the permanent object. In this case is it true to say `Temporary tables and views can shadow views. If the view has a shadow, in the current session, then the dropView will fail as the view will not be accessible.`

Reply:
yes, isn't it covered by 
>If a permanent object in a given path exists,
        it will be inaccessible in the current session
        
?

Reply:
Yes it is covered - I was looking to be specific and use the word view or table where every we can for clarity - rather than use object. I think as it is quite complicated - trying to be as explicit as possible would help the reader.
For example my assumptions are:
- dropView I assume only could drop a view, not a temporary view, a connector table or temporary table. So using object is not very obvious. 
- dropTable  I assume dropTable means attempt to drop a table, and should not be used against temporary objects.  
 (can dropTable be used against a view, as a view is a table or not).

Reply:
The problem with such approach is that at some point there might appear another temporary entity.
Then need to go through all the docs we have and fix it everywhere.
As an example recently came `Materialized Table` FLINK-35187, luckily not temporary (yet).